HCore-Tenants/Providers/Impl/AuditProviderImpl.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using HCore.Audit.Models;
+using HCore.Audit.Providers;
+
+namespace HCore.Tenants.Providers.Impl
+{
+    internal class AuditProviderImpl : IAuditProvider
+    {
+        private readonly ITenantDataProvider _tenantDataProvider;
+
+        public AuditProviderImpl(ITenantDataProvider tenantDataProvider)
```

Comment:
This does not always work! See my comment about this! There is also IAMQPTenantDataProvider - see ConfiguredComponentProvider.cs, line 1522 ff. - you can move IAMQPTenantDataProvider to HCore to support this scenario.

Reply:
I assume you meant `ITenantInfoAMQPAccessor`